libbeat/outputs/kafka/client.go

```diff
@@ -113,7 +113,7 @@ func newKafkaClient(
 	return c, nil
 }
 
-func (c *client) Connect() error {
+func (c *client) Connect(_ context.Context) error {
```

Comment:
If you revert this change and run the integration test, it will fail with the panic in the original issue.